invoke-ai/InvokeAI
# fix(ui): fix missing images not handled

## What type of PR is this? (check all applicable)


## Description

fix(ui): fix missing images not handled

- Reset init image, control adapter images, and node image fields when their selected image fails to load
- Only do this if the app is connected via socket (this indicates that the image is "really" gone, and there isn't just a transient network issue)

It's possible for image parameters/nodes/states to have reference a deleted image. For example, a resize image node might have an image set on it, and the workflow saved. The workflow contains a hard reference to that image.

The image is deleted and the workflow loaded again later. The deleted image is still in that workflow, but the app doesn't detect that. The result is that the workflow/graph appears to be valid, but will fail on invoke.

This creates a really confusing user experience, where when somebody shares a workflow with an image baked into it, and another person opens it, everything *looks* ok, but the workflow fails with a mysterious error about a missing image.

The problem affects node images, control adapter images and the img2img init image. Resetting the image when it fails to load *and* socket is connected resolves this in a simple way.

The problem also affects canvas images, but we have handle that by displaying an error fallback image, so no change is made there.


## Related Tickets & Documents



- Closes #5205

## QA Instructions, Screenshots, Recordings

Follow the repro in the linked issue. You can also try:
- setting your init and/or a controlnet image
- ensure the invoke button is active, indicating there are no problems w/ your current settings
- from a diff browser, delete those images
- restart the server to clear the internal image cache
- refresh the your original browser window
- note that the invoke button is still lit up, and that invoking fails

On this PR, once you refresh, the images should all reset and the invoke button should be disabled, with the tooltip saying missing init/control/node image.
